File: Problem2/gemini1.java
import java.util.Scanner;

public class gemini1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        if (sc.hasNextInt()) {
            int n = sc.nextInt();
            long totalSum = 0;
            
            for (int i = 0; i < n; i++) {
                totalSum += sc.nextLong();
            }
            
            // If the sum is divisible by n, it's possible to equalize
            if (totalSum % n == 0) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }
        sc.close();
    }
}

File: Problem2/gemini1.py
import sys

def solve():
    # Read all inputs from standard input
    input_data = sys.stdin.read().split()
    
    if not input_data:
        return
        
    n = int(input_data[0])
    
    # Calculate the total sum of the array
    # input_data[1:] contains the elements of the array
    total_sum = sum(int(x) for x in input_data[1:])
    
    # If the sum is divisible by n, it's possible to equalize
    if total_sum % n == 0:
        print("YES")
    else:
        print("NO")

if __name__ == '__main__':
    solve()